aiosip/dialog.py

```diff
@@ -149,3 +206,68 @@ def register(self, headers=None, attempts = 3):
                                             headers=headers,
                                             payload='')
         return send_msg_future
+
+    @asyncio.coroutine
+    def invite(self, headers=None, sdp=None, attempts=3):
+        if self.invite_current_attempt:
+            raise RegisterOngoing('Already a invite going on ! (attempt %s)'%self.register_current_attempt)
+
+        self.invite_current_attempt = attempts
+        ok = yield from self.send_message(method='INVITE',
+                                          headers=headers,
+                                          payload=sdp)
+        assert ok.status_code == 200
+
+        ok_cseq = int(ok.headers['CSeq'].split()[0])
+        self.cseqs['ACK'] = ok_cseq - 1
+        self.cseqs['BYE'] = ok_cseq
+
+        def send_ack(ok):
+            hdrs = multidict.CIMultiDict()
+            hdrs['Via'] = headers['Via']
+            hdrs['From'] = ok.headers['From']
+            hdrs['To'] = ok.headers['To']
+            hdrs['CSeq'] = ok.headers['CSeq'].replace('OK', 'ACK')
+            self.send_message(method='ACK', headers=hdrs)
+
+        class Call:
```

Comment:
Yeah, I can address that. Its inside the coroutine because closes over the outside scope.